### DISCUSSION TITLE
blog/retrospect/1-devcourse-MIL-1

### DISCUSSION THREAD

**Discussion by giscus** [pySoo/Next-blog#72] [General]
# blog/retrospect/1-devcourse-MIL-1

퇴사 후 스스로 느꼈던 아쉬웠던 점들을 극복하기 위해서 시도했던 경험들과 그 과정에서 배우고 느낀 점들을 기록합니다.

https://enjoydev.life/blog/retrospect/1-devcourse-MIL-1

**Comment by SeungHyune** [comment]
수현님 안녕하세요 !!
함께 일하고 싶은 개발자 되기를 진행 중으로 해주셨는데.. 저는 수현님과 같은 개발자와 일하면 정말 좋을 거 같다는 생각을 하고 있습니다 ! ㅎㅎ
그렇다는 건 이미 함께 일하고 싶은 개발자가 되신 것 아닐까요??! 😊

지금도 너무 잘하고 계시지만 앞으로가 더 기대되는 개발자 이신 것 같습니다. 
(제가 느낄 때 항상 완벽하다고 느꼈었는데, 그런 고민을 하고 계실 줄은 몰랐습니다.. 저는 그 부분이 스트레스이실 수도 있을 거라고 생각하지만, 수현님의 장점 중 하나가 아닐까 하는 생각을 하게 되네요 !! 지금은 바쁜 준비과정을 보내고 있다 보니까 더 그렇게 느끼실 것 같지만..! 나중에 조금 바쁜 게 지나간다면 그 부분은 굉장한 장점이라고 생각합니다 !!)
우선 몸이 우선이니까 하루빨리 회복하셨으면 좋겠습니다 😉

**Reply by pySoo** [reply]
헉 승현님 언제 이런 정성스러운 댓글을... 문장 문장 힘이 되네요 감사합니다 😭
저는 아직도 갈 길이 멀다고 느끼지만..!!! 좋게 봐주셔서 감사합니다 ㅎㅎ

말씀대로 어쩌면 지금은 여유가 없어서 그렇게 느껴지는 것 같기도 하네요!
스스로를 채찍질할 때마다 해주신 말씀들 떠올리면서 스트레스보다는 좋은 방향으로 활용해 보려고 노력해보겠습니다 ㅎㅎ
저도 승현님과 팀을 하면서 많이 성장해서 팀원분들 너무 잘 만났다는 생각을 하고 있었습니다 😁 앞으로도 계속 열심히 해봐요!!! 🔥🔥🔥

**Comment by Seung-wan** [comment]
어디선가 제 얘기가...
수현님 퇴사하시구 또 다른 도전을 하고 계셨군요!
재밌게 잘 읽었어요~!

**Reply by pySoo** [reply]
헉!!!! 승완님께서 이 글을 읽으실 거라곤 생각 못했었는데... 🤣 부끄럽네요..
로토님 스터디에서 꼼꼼히 리뷰 해주신 덕분에 많이 배웠습니다!!! (제 코드 리뷰 스승님이십니다요)

승완님께 리뷰 받으면서 이렇게도 생각할 수 있구나.. 하면서 노션에 따로 기록해두었는데
지금 저도 동료분들 코드 리뷰 하면서 정말 유용하게 사용하고 있어요!
감사하다는 말을 못드려서 언젠가는 전하고 싶었는데 이렇게 전할 수 있어서 기쁩니다 🤩

하나 여쭤보고 싶었던게 있었는데.. 승완님은 코드에 대한 가치관이나 기준을 어떻게 세우게 되셨나요??
주관있게 리뷰하시는 분들을 보면서 정말 배우고 싶었어요!

**Reply by Seung-wan** [reply]
감사합니다..! 저도 같이 스터디하면서 활발하게 얘기 나눌 수 있어서 재밌었어요 😁
데브코스도 엄청 재밌어보이네요! (최근에 쿼타랩 면접을 보고 왔는데 동욱님팀이어서 신기했어요 ㅋㅋㅋ)
인프콘에서 인사드리려고 했었는데 이전 직장 동료분들을 만나가지구 마지막 세션은 아예 듣지도 못했네요 ㅠㅠ...

저는 회사나 교육 프로그램에서 다양한 개발자분들과 코드 리뷰를 많이 했었는데요. 이 과정속에서 느낀점은 코드에 정답은 없지만, 
어느 정도의 원칙은 존재한다는 것이었어요. 간단하게는 DRY, YAGNI와 같은 원칙들이 있을 것 같아요.
그래서 저는 많은 개발자들이 옳다고 생각하는 대원칙같은 것들은 메인으로 가져가면서, 그 외의 개발자들마다 의견이 갈리는 부분들은
제가 생각했을 때 맞다고 생각하는 기준을 세웠던 것 같아요.
그리고 이런 기준을 세우기 위해서 꾸준히 책을 읽고 코드를 작성하고 코드 리뷰를 하면서 제 생각을 정리하는데 도움이 됐던 것 같네요.
또 리뷰를 남기기 전에도 저도 한번 더 생각을 정리하고 독스를 찾아보면서 공부하고 남기는 편이에요!

**Reply by pySoo** [reply]
우와 ㅋㅋㅋㅋ 신기하네요 세상 참 좁은 것 같아요
저도 인프콘에서 인사 드리고 싶었는데 인파에 밀려서 강연만 듣고 바로 나왔던 것 같아요.. 😂

오.. 대원칙을 메인으로 두고 그 외의 것들은 승완님만의 기준을 세우셨군요!
저는 요즘 모르는게 있으면 GPT를 바로 찾는 편인데... 편안함에 안주하지 말고 책이랑 공식 문서들도 꾸준히 찾아봐야겠어요!

요즘 코드 리뷰하면서 의견 나누는게 넘 재밌어서 더 잘하고 싶다는 생각이 들었는데
말씀해주셔서 큰 도움이 될 것 같아요 ㅎㅎ
노하우들 공유해주셔서 감사합니다!!! 👍👍👍

**Comment by evan-moon** [comment]
안녕하세요 수현님. 본인에 대한 성장 의지와 동시에 다른 동료분들까지 챙겨주시려고 노력하시는 모습이 정말 보기 좋았습니다.

비록 제가 멘토이긴 했어도 저도 아직 부족한 점이 많아요. 저는 스스로에게든 타인에게든 비판적인 성격이라 타인의 장점을 바라보고 칭찬하는 역량이 많이 부족합니다 ㅎㅎ 머리로는 칭찬이 필요하다는 걸 알아도 잠깐만 방심하면 개선해야할 점에 더 집중하고 있더라구요.

9주라는 짧은 기간이었지만 수현님께서 동료분들을 챙겨주시고 코드리뷰에서도 동료들이 잘 작성한 부분을 짚어주시는 모습을 보고 저 또한 많은 영감을 받았습니다.
지난 번 점심먹으면서 나눴던 이야기처럼 수현님께서 이런 역량을 앞으로도 유지하고 성장하실 수 있다면, 4-5년 뒤에는 좋은 리더십을 가진 개발자가 될 수 있을 것이란 기대가 됩니다.

**Reply by pySoo** [reply]
으아 멘토님 제 블로그를 방문해 주시다니...! 정말 영광입니다 🤩🥺
데브코스의 시작을 멘토님과 저희 팀원분들과 할 수 있어서 참 감사했고 덕분에 정말 많이 배웠습니다!
좋은 토론 주제를 공유해주시는 멘토님과 의견 공유를 좋아하는 저희 팀원들 성향이 잘 맞아서 저희 팀 시너지가 더 좋았던 것 같아요 ㅎㅎ

주니어 시기에 했던 고민들 잘 간직하고 앞으로 풀어가면서, 4년 후에는 멘토님처럼 좋은 리더가 될 수 있도록 꾸준히 노력하겠습니다!
그리고 오히려 전 가끔씩 멘토님께서 해주시는 칭찬 멘트가 정말 짜릿하더라구요.. 🤭
칭찬을 많이 하는 성향이 아니시더라도 한 번 해주실 때 진정성이 느껴져서 그런지 더 감동이었습니다.. ㅎㅎ

이번 팀에서 배웠던 내용 잘 간직해서 앞으로의 개발 여정에서도 잘 활용해보도록 하겠습니다! 감사해요 🙇‍♀️

**Comment by kyechan99** [comment]
안녕하세요!!
우연히 작업하신 노션 코드를 보다가 이렇게 블로그까지 닿게 되었네요 :smile:

저는 수현님 코드를 보는 일이 재미있고 계속 파헤치게 하는 느낌이 들었어요.
그런 느낌이 들게 하는 코드는 어떤 코드보다 제가 지향하는 코드였거든요.
형태는 다르지만, 비슷한 작업을 했었기 때문일까요?
어떤 생각과 고민으로 코드와 스타일을 구상해냈을까 궁금하게 된 것 같아요.
그런면에서 함께 일해보고 싶고 즐거울 것 같은 분이라고 생각되었습니다.

블로그 포스팅도 너무 재밌게 읽었어요. :thumbsup: :thumbsup:
카테고리 구성을 어떻게 생각해내신 건지 멋지네요 !!

**Reply by pySoo** [reply]
우와 안녕하세요 예찬님 😄
제 코드를 읽고 그런 느낌을 받으셨다니…!
개발자로서 들을 수 있는 최고의 칭찬을 들은 것 같아서 정말 기쁘고 감사하네요 😭

저는 제가 생각했을 때 잘 만들었다고 느꼈던 다른 분들의 코드들을 기록해두고 자주 참고하는 편이에요!
아무래도 이미 다른 분들이 충분히 고민하셨던 내용에 제 생각이 추가된 형태로 코드가 만들어지다보니
재밌다는 인상을 받으셨을 수 있겠다는 생각이 드네요 ㅎㅎ
블로그도 재미있게 읽어주셨다니 너무 감사합니다. (제가 가장 좋아하는 칭찬이에요 ☺️)

저는 예찬님 git을 둘러보다가 정말 놀랐어요!
예찬님이 만드신 **`capsule-render`** 오픈소스 알고보니 제 readme에서도 사용하고 있더라구요
제가 개발자로서 이루고 싶은 장기적인 목표에 사용자들이 꾸준히 사용하는 오픈소스를 개발하는 것이 있는데
이미 예찬님은 그 부분을 이루신 것 같아서 정말 멋지다는 생각이 들었습니다 👍

방문해주셔서 너무 감사드려요! 
저도 종종 예찬님의 블로그와 깃허브에 방문해서 영감 받아가겠습니다 ㅎㅎ

**Comment by jjhangu** [comment]
블로그가 아주 심플하고, 잘 작성되어있어서 글 남기고 갑니다. 
github도 방문했었는데 전체적인 분위기에 압도되어(깔끔, 정돈, 강조, 이모티콘등) 좋은 기분으로 많을걸 배우고 갑니다 👍

**Reply by pySoo** [reply]
안녕하세요!
아직 더 발전시킬 부분들이 많지만... 좋은 인상을 받으셨다니 정말 기쁩니다 😊
앞으로도 이해하기 쉬운 글들을 작성하기 위해서 꾸준히 노력하겠습니다 ㅎㅎ 방문해주셔서 감사해요! 👍
